app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRPresenter.java

```diff
@@ -0,0 +1,116 @@
+package org.fossasia.openevent.app.qrscan;
+
+import android.util.SparseArray;
+
+import com.google.android.gms.vision.barcode.Barcode;
+
+import org.fossasia.openevent.app.data.contract.IEventDataRepository;
+import org.fossasia.openevent.app.data.models.Attendee;
+import org.fossasia.openevent.app.qrscan.contract.IScanQRPresenter;
+import org.fossasia.openevent.app.qrscan.contract.IScanQRView;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+import io.reactivex.subjects.PublishSubject;
+
+public class ScanQRPresenter implements IScanQRPresenter {
+
+    private long eventId;
+
+    private IScanQRView scanQRView;
+    private IEventDataRepository eventDataRepository;
+    private List<Attendee> attendees = new ArrayList<>();
+
+    private PublishSubject<Boolean> detect = PublishSubject.create();
+    private PublishSubject<String> data = PublishSubject.create();
+
+    public ScanQRPresenter(long eventId, IScanQRView scanQRView, IEventDataRepository eventDataRepository) {
+        this.eventId = eventId;
+        this.scanQRView = scanQRView;
+        this.eventDataRepository = eventDataRepository;
+
+        detect.distinctUntilChanged()
+            .debounce(150, TimeUnit.MILLISECONDS)
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(receiving -> scanQRView.showInView(!receiving));
+
+        data.distinctUntilChanged()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(this::processBarcode);
+    }
+
+    private void processBarcode(String barcode) {
+        scanQRView.showData(barcode);
+
+        Observable.fromIterable(attendees)
+            .filter(attendee -> (attendee.getOrder().getIdentifier() + "-" + attendee.getId()).equals(barcode))
```

Comment:
This can be done -> Create `AttendeeModel`, use it to 
- Toggle Status Check
- Match against barcode (Not really sure though why we need it out of QR Presenter)

If we do this, we move out toggleAttendee from EventRepository because that too isn't the job of a repository